### PR TITLE
integrate with dbass: support embed dasbhoard by iframe

### LINKED ISSUE #535: Customization of dashboard when used as iframe
## Feature Request

**Is your feature request related to a problem? Please describe:**

**Describe the feature you'd like:**

use iframe' src url params to indicate : 

 - hide the side-menu(headless)
 - inject auth token to dashboard

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

### PR DESCRIPTION
**Pull request by baurine**
resolve #535 

What did:

- Support assign token/hideNav/lang params by URL

Demo:

### CODE REVIEW

File: ui/lib/utils/appOptions.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import * as auth from '@lib/utils/auth'
+import * as i18n from '@lib/utils/i18n'
+
+export interface AppOptions {
+  token: string | null
+  hideNav: boolean
```
Comment (baurine) [inline ui/lib/utils/appOptions.ts:6]: I think `headless` is a little confusing, so I change it to `hideNav` to make it more specific, or we have some better naming.

File: ui/dashboardApp/index.js [hunk 2]
```diff
@@ -34,13 +37,17 @@ async function main() {
...
+      'root'
+    ),
     () => {
       return !routing.isLocationMatchPrefix(auth.signInRoute)
     },
     { registry }
   )
 
+  // FIXME: Should we mount signin page when embeded by iframe?
   singleSpa.registerApplication(
```
Comment (baurine) [inline ui/dashboardApp/index.js:51]: Should we mount the sign-in page when embedded by iframe?

**Comment by baurine** [comment]
Hi @breeswish @sivagao , this feature is almost done except to handle the prefix path, please help take a look whether the solution is available when you're free.

The app options are stored in the memory instead of session storage while they both work.

**Comment by baurine** [comment]
According to the PR #504 , the prefix path is already handled by the backend and passed to the frontend, we don't need to care about it now, right?

File: ui/public/iframe-test.html [hunk 3]
```diff
@@ -9,9 +9,9 @@
       function changeApp(app, lang) {
```
Comment (breeswish) [inline ui/public/iframe-test.html:9]: This file will be packed and embedded into PD. Please remove it.

Comment (baurine) [reply ui/public/iframe-test.html:9]: Got it, will remove it later.

**Comment by baurine** [comment]
The html to test:

```html
<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
    <link rel="icon" href="/favicon.ico" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <title>TiDB Dashboard iframe Test</title>
    <script>
      function changeApp(app, lang) {
        console.log('app:', app)
        const dashboardUrl = 'http://127.0.0.1:3000/dashboard#/'
        const token =
          'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE1OTE3Njc4NzcsIm9yaWdfaWF0IjoxNTkxNjgxNDc3LCJwIjoiQUxteXFyTGZFd0ljR01UcU8wdVR1SlJhUFdVQWRBMGZrUjFEMTFvRFcrbWJ3ZWQ0dnFJVjFFT1V4WXlpekhUL0tmdzVqYi9sV1dtVDdvUWRYUWV3bnpXdndaTnZDaDJUM1o4RmR5dUdOeUdPMXFsTHZIeVoifQ.j0Gjx06VhZAGY1Q1vCoWpj-bw3VWbrVyMkRkYehV9V8'
        const fullUrl = `${dashboardUrl}${app}?token=${token}&hideNav=true&lang=${
          lang || 'en'
        }`
        document.getElementById('dashboard').src = fullUrl
      }
      window.onload = function () {
        changeApp('statement')
      }
    </script>
  </head>
  <body>
    <div style="margin: 20px;">
      <h1>iframe test</h1>
      <div>
        <a href="#" onclick="changeApp('statement'); return false;"
          >Statement</a
        > | 
        <a href="#" onclick="changeApp('slow_query'); return false;"
          >Slow Query</a
        >
      </div>
      <div style="border: 1px solid #52c41a; height: 80vh;">
        <iframe id="dashboard" width="100%" height="100%" />
      </div>
    </div>
  </body>
</html>
```

File: ui/public/iframe-test.html (hunk 3 above)
Comment (baurine) [reply ui/public/iframe-test.html]: Done

**Comment by baurine** [comment]
close it because it is implemented by #628

**Pull request closed without merging**